pkg/inputs/snmp/x/meraki/meraki.go

```diff
@@ -0,0 +1,685 @@
+	deviceSet := map[string][]networkDevice{}
+	for _, network := range c.networks {
+		params := networks.NewGetNetworkDevicesParams()
+		params.SetNetworkID(network.ID)
+
+		prod, err := c.client.Networks.GetNetworkDevices(params, c.auth)
+		if err != nil {
+			return nil, err
+		}
+
+		b, err := json.Marshal(prod.GetPayload())
+		if err != nil {
+			return nil, err
+		}
+
+		var devices []networkDevice
+		err = json.Unmarshal(b, &devices)
+		if err != nil {
+			return nil, err
+		}
+
+		for i, _ := range devices {
+			if devices[i].Name == "" {
+				devices[i].Name = devices[i].Serial
+			}
+			devices[i].network = network
+		}
+
+		if len(devices) > 0 {
+			deviceSet[network.Name] = devices
+		}
+	}
+
+	return deviceSet, nil
+}
+
+func (c *MerakiClient) getDeviceClients(dur time.Duration) ([]*kt.JCHF, error) {
+	networkDevs, err := c.getNetworkDevices()
+	if err != nil {
+		return nil, err
+	}
+
+	c.log.Infof("Got devices for %d networks", len(networkDevs))
+	clientSet := []client{}
+	durs := float32(3600) // Look back 1 hour in seconds, get all devices using APs in this range.
+	for network, deviceSet := range networkDevs {
+		c.log.Infof("Looking at %d devices for network %s", len(deviceSet), network)
+		for _, device := range deviceSet {
+			params := devices.NewGetDeviceClientsParams()
+			params.SetSerial(device.Serial)
+			params.SetTimespan(&durs)
+
+			prod, err := c.client.Devices.GetDeviceClients(params, c.auth)
+			if err != nil {
+				return nil, err
+			}
+
+			b, err := json.Marshal(prod.GetPayload())
+			if err != nil {
+				return nil, err
+			}
+
+			var clients []client
+			err = json.Unmarshal(b, &clients)
+			if err != nil {
+				return nil, err
+			}
+			for _, client := range clients {
+				client.network = network
+				client.device = device
+				clientSet = append(clientSet, client) // Toss these all in together
+			}
+		}
+	}
+
+	return c.parseClients(clientSet)
+}
+
+func (c *MerakiClient) parseClients(cs []client) ([]*kt.JCHF, error) {
+	res := make([]*kt.JCHF, 0)
+	for _, client := range cs {
+		dst := kt.NewJCHF()
+		if client.IPv6 != "" {
+			dst.DstAddr = client.IPv6
+		} else {
+			dst.DstAddr = client.IP
+		}
+		dst.CustomStr = map[string]string{
+			"network":            client.network,
+			"client_id":          client.ID,
+			"description":        client.Description,
+			"status":             client.Status,
+			"vlan_name":          client.NamedVlan,
+			"client_mac_addr":    client.Mac,
+			"user":               client.User,
+			"manufacturer":       client.Manufacturer,
+			"device_type":        client.DeviceType,
+			"recent_device_name": client.RecentDeviceName,
+			"dhcp_hostname":      client.DhcpHostname,
+			"mdsn_name":          client.MdnsName,
+		}
+		dst.CustomInt = map[string]int32{
+			"vlan": int32(client.Vlan),
+		}
+		dst.CustomBigInt = map[string]int64{}
+		dst.EventType = kt.KENTIK_EVENT_SNMP_DEV_METRIC
+		//dst.Provider = c.conf.Provider // @TODO, pick a provider for this one.
```

Comment:
do we need to have a provider?

Reply:
This is a NR specific thing. They don't right now know what provider to set so they asked to remove this for now entirely.